sql/hive-thriftserver/src/main/scala/org/apache/spark/sql/hive/thriftserver/HiveThriftServer2.scala

```diff
@@ -57,12 +60,15 @@ object HiveThriftServer2 extends Logging {
     logInfo("Starting SparkContext")
     SparkSQLEnv.init()
 
-    Runtime.getRuntime.addShutdownHook(
-      new Thread() {
+    // Clean up after we exit. Use higher priority than FileSystem.
+    assert(SHUTDOWN_HOOK_PRIORITY > FileSystem.SHUTDOWN_HOOK_PRIORITY)
```

Comment:
No, it doesn't. In fact, ShutdownHookManager does not exist in Hadoop 1.x. So we can't do this, can we? I actually tried to compile Spark with Hadoop 1.x, and it failed with the following errors-

-----
object ShutdownHookManager is not a member of package org.apache.hadoop.util
value SHUTDOWN_HOOK_PRIORITY is not a member of object org.apache.hadoop.fs.FileSystem
-----

So we're back to ground zero. :(